test/common/shared_functions.go

```diff
@@ -217,3 +220,65 @@ func WriteRHMICRToFile(client dynclient.Client, file string) error {
 		return writeObjToYAMLFile(rhmi, file)
 	}
 }
+
+// Common function to perform CRUDL and verifying their expected permissions
+func verifyCRUDLPermissions(t *testing.T, openshiftClient *resources.OpenshiftClient, expectedPermission ExpectedPermissions) {
+	// Perform LIST Request
+	resp, err := openshiftClient.DoOpenshiftGetRequest(expectedPermission.ListPath)
+
+	if err != nil {
+		t.Errorf("failed to perform LIST request with error : %s", err)
+	}
+
+	if resp.StatusCode != expectedPermission.ExpectedListStatusCode {
+		t.Errorf("unexpected response from LIST request : %v", resp)
+	}
+
+	// Perform CREATE Request
+	bodyBytes, err := json.Marshal(expectedPermission.ObjectToCreate)
+
+	resp, err = openshiftClient.DoOpenshiftPostRequest(expectedPermission.ListPath, bodyBytes)
```

Comment:
Assuming the post request fails, will the `get`, `update` and `delete` functions still work? If not does it make sense just to return here?

Reply:
Yeah, the `get`, `update` and `delete` should still work even on failure here. `t.Fatalf` would immediately stop the code execution and fail the test, but `t.Errorf` would continue the code execution and log the test as a failure at the end I think :thinking:

Reply:
Ok :+1: